metrics_collector/instance_manager_collector.go

```diff
@@ -188,3 +211,40 @@ func (imc *InstanceManagerCollector) collectRequestValues(ch chan<- prometheus.M
 		ch <- prometheus.MustNewConstMetric(imc.memoryRequestMetric.Desc, imc.memoryRequestMetric.Type, requestMemoryBytes, imc.currentNodeID, pod.GetName(), instanceManagerType)
 	}
 }
+
+func (imc *InstanceManagerCollector) collectGrpcConnection(ch chan<- prometheus.Metric) {
+	defer func() {
+		if err := recover(); err != nil {
+			imc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	engineInstanceManagers, err := imc.ds.ListInstanceManagersByNode(imc.currentNodeID, longhorn.InstanceManagerTypeEngine)
+	if err != nil {
+		imc.logger.WithError(err).Warn("error during scrape")
+		return
+	}
+
+	for _, im := range engineInstanceManagers {
+		imPod, err := imc.ds.GetPod(im.Name)
+		if err != nil {
+			if datastore.ErrorIsNotFound(err) {
+				logrus.WithError(err).Debugf("resetting proxy gRPC connection counter for %v", im.Name)
+				imc.proxyConnCounter.ResetCount()
```

Comment:
When having 2 IM-E pods after Longhorn upgrade, there will be two cases:
- No differentiated metrics for each pod, so the reported metrics actually the same for them
- If the first pod not found, the counter will be reset so it will cause the other reported metrics probably incorrect

However, this can be improved later.

Reply:
https://github.com/longhorn/longhorn/issues/4112